app/auth/middleware.go

```diff
@@ -33,11 +34,32 @@ func Middleware(provider Provider) mux.MiddlewareFunc {
 	}
 }
 
+// LegacyMiddleware tries to authenticate user using request header
+func LegacyMiddleware(provider Provider) mux.MiddlewareFunc {
```

Comment:
Not much difference between `LegacyMiddleware` and `Middleware` as well.

Reply:
Yeah, this was specific choice to minimize risk. I suspect the `LegacyMiddleware` will be completely removed in a few months, so to make sure it was a cleaner split I just created a new middleware and named the old one Legacy.